cupy/random/permutations.py

```diff
@@ -1,7 +1,23 @@
-# flake8: NOQA
-# "flake8: NOQA" to suppress warning "H104  File contains nothing but comments"
+import cupy
+from cupy.random import generator
 
-# TODO(okuta): Implement shuffle
+# TODO(okuta): Implement permutation
 
 
-# TODO(okuta): Implement permutation
+def shuffle(a):
+    """Shuffles an array.
+
+    Args:
+        a (cupy.ndarray): The array to be shuffled.
+
+    .. seealso:: :func:`numpy.random.shuffle`
+
+    """
+    if not isinstance(a, cupy.ndarray):
+        raise TypeError('The array must be cupy.ndarray')
+
+    if a.ndim == 0:
```

Comment:
Users can call `RandomState.shuffle` directly. Move these codes to `generator.py`.